bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+                    setSoundProgram(command.toString(), zone);
+                    break;
+                case CHANNEL_SELECTPRESET:
+                    setPreset(command.toString(), zone);
+                    break;
+                case CHANNEL_PLAYER:
+                    if (command.equals(PlayPauseType.PLAY)) {
+                        setPlayback("play");
+                    } else if (command.equals(PlayPauseType.PAUSE)) {
+                        setPlayback("pause");
+                    } else if (command.equals(NextPreviousType.NEXT)) {
+                        setPlayback("next");
+                    } else if (command.equals(NextPreviousType.PREVIOUS)) {
+                        setPlayback("previous");
+                    } else if (command.equals(RewindFastforwardType.REWIND)) {
+                        setPlayback("fast_reverse_start");
+                    } else if (command.equals(RewindFastforwardType.FASTFORWARD)) {
+                        setPlayback("fast_forward_end");
+                    }
+                    break;
+                case CHANNEL_SLEEP:
+                    setSleep(command.toString(), zone);
+                    break;
+                case CHANNEL_MCSERVER:
+                    action = "";
+                    json = "";
+                    if (command.toString().equals("")) {
+                        action = "unlink";
+                        role = "";
+                        groupId = "";
+                    } else {
+                        action = "link";
+                        String[] parts = command.toString().split("\\*\\*\\*");
+                        mclinkSetupServer = parts[0];
+                        mclinkSetupZone = parts[1];
+                        tmpString = getDistributionInfo(mclinkSetupServer);
+                        @Nullable
+                        DistributionInfo targetObject = new Gson().fromJson(tmpString, DistributionInfo.class);
+                        responseCode = targetObject.getResponseCode();
+
+                        role = targetObject.getRole();
+                        if (role.equals("server")) {
+                            groupId = targetObject.getGroupId();
+                        } else if (role.equals("client")) {
+                            groupId = "";
+                        } else if (role.equals("none")) {
+                            groupId = generateGroupId();
+                        }
+                    }
+
+                    if (action.equals("unlink")) {
+                        json = "{\"group_id\":\"\"}";
+                        httpResponse = setClientInfo(this.host, json);
+                    } else if (action.equals("link")) {
+                        json = "{\"group_id\":\"" + groupId + "\", \"zone\":\"" + mclinkSetupZone
+                                + "\", \"type\":\"add\", \"client_list\":[\"" + this.host + "\"]}";
+                        logger.info("setServerInfo json: {}", json);
```

Comment:
See above logging to info. Please check all.

Reply:
Done